File: cpp/643_maximum_average_subarray_i.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
		int sum = 0;
		for (int i = 0; i < k; i++) {
			sum += nums[i];
		}
		
		int big = sum;
		for (int i = k; i < nums.size(); i++) {
			sum += nums[i];
			sum -= nums[i - k];
			if (sum > big) {
				big = sum;
			}
		}
		return ((double) big) / k;
    }
};

File: go/643_maximum_average_subarray_i.go
package main

func findMaxAverage(nums []int, k int) float64 {
	sum := 0.0
	for i := range k - 1 {
		sum += float64(nums[i])
	}

	max_avg := -100000.0
	for i := range len(nums) - k + 1 {
		sum += float64(nums[k+i-1])
		max_avg = max(max_avg, sum/float64(k))
		sum -= float64(nums[i])
	}

	return max_avg
}
